lib/connectors/SentinelConnector/types.ts

```diff
@@ -1,4 +1,32 @@
+import { IRedisOptions } from "../../redis/RedisOptions";
+
 export interface ISentinelAddress {
   port: number;
   host: string;
+  family?: number;
+}
+
+// TODO: A proper typedef. This one only declares a small subset of all the members.
+export interface IRedisClient {
+  options: IRedisOptions;
+  sentinel(subcommand: "sentinels", name: string): Promise<string[]>;
+  sentinel(
+    subcommand: "get-master-addr-by-name",
+    name: string
+  ): Promise<string[]>;
+  sentinel(subcommand: "slaves", name: string): Promise<string[]>;
+  subscribe(...channelNames: string[]): Promise<number>;
+  on(
+    event: "message",
+    callback: (channel: string, message: string) => void
+  ): void;
+  on(event: "error", callback: (error: Error) => void): void;
+  on(event: "reconnecting", callback: () => void): void;
+  disconnect(): void;
+}
+
+export interface ISentinel {
+  address: Partial<ISentinelAddress>;
+  isConnected: boolean;
+  getClient: () => IRedisClient;
```

Comment:
We already support the `lazyConnect` option which I think has a similar purpose. With lazyConnect, clients will connect to the server only at the first time they need to send a command. Seems this can save `isConnected` and `getClient`.